### PR TITLE
[backend] Fix #139 and #142

### LINKED ISSUE #139: No tags in `/api/product/:id`
**Describe the bug**
As Title

### LINKED ISSUE #142: No coupon detail API exists
**Describe the bug**
There is no coupon details API exist for customer

### PR DESCRIPTION
**Pull request by Error0229**
Resolves #139 Resolves #142
the current response for  `/api/product/:id` is 
```json
{
  "description": "string",
  "expire_date": "string",
  "id": 0,
  "name": "string",
  "price": 0,
  "product_image_url": "string",
  "sales": 0,
  "seller_name": "string",
  "shop_image_url": "string",
  "shop_name": "string",
  "stock": 0,
  "tags": [
    {
      "id": 0,
      "name": "string"
    }
  ]
}
```
the current response for `/api/shop/{seller_name}/coupon/{id}` like under
```json
{
  "description": "string",
  "discount": 0,
  "expire_date": "string",
  "id": 0,
  "name": "string",
  "scope": "global",
  "start_date": "string",
  "tags": [
    {
      "id": 0,
      "name": "string"
    }
  ],
  "type": "percentage"
}
```
### Changes
- Add `tags` for `/api/product/:id`
- Create `/api/shop/{seller_name}/coupon/{id}` endpoint
- Add additional error check at `pkg/router/seller/shop.go` `GetShopInfo`
### Testing Approach (if unconventional)

Call `/api/product/:id` with any exist and enable product's id
It should include corresponding product's  `tags` in response.
Create any coupon for shop and call `/api/shop/{seller_name}/coupon/{id}` with its `id` and `seller_name`
Also, you may use admin to create a global coupon, and call the same API, and the `"tags"` in response would be `[]`

### CODE REVIEW

**Review by UGisBusy — DISMISSED** [review]
womm

**Review by UGisBusy — APPROVED** [review]
womm

**Review by UGisBusy — APPROVED** [review]
this count if i review again?

File: pkg/router/general/shop.go
Comment (NOOBDY) [inline pkg/router/general/shop.go:1]: would this also break if there are no rows

Comment (UGisBusy) [reply pkg/router/general/shop.go:1]: if u mean the coupon has no row or have no coupon.
no. it works great ( at least no error )

**Review by NOOBDY — APPROVED** [review]

**Pull request merged**